clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -92,39 +102,45 @@ object UncommittedGarbageCollector {
       // Process uncommitted
       val uncommittedGCRunInfo =
         new APIUncommittedAddressLister(apiClient).listUncommittedAddresses(spark, repo)
+
       var uncommittedDF =
-        if (uncommittedGCRunInfo.uncommittedLocation != "")
+        if (uncommittedGCRunInfo.uncommittedLocation != "") {
           spark.read.parquet(uncommittedGCRunInfo.uncommittedLocation)
-        else {
+        } else {
           // in case of no uncommitted entries
           spark.emptyDataFrame.withColumn("physical_address", lit(""))
         }
       uncommittedDF = uncommittedDF.select(uncommittedDF("physical_address").as("address"))
       runID = uncommittedGCRunInfo.runID
 
       // Process committed
-      val committedDF =
-        new NaiveCommittedAddressLister().listCommittedAddresses(spark, storageNamespace)
+      val committedDF = new NaiveCommittedAddressLister()
+        .listCommittedAddresses(spark, storageNamespace)
 
       addressesToDelete = dataDF
         .except(committedDF)
         .except(uncommittedDF)
+
       // TODO (niro): not working - need to find the most efficient way to save the first slice
-      val firstFile = dataDF.select(col("address")).first().toString()
-      firstSlice = firstFile.substring(0, firstFile.lastIndexOf("/"))
+      if (!dataDF.isEmpty) {
```

Comment:
This will still not work since dataDF contains both "old" and "new" address so basically the first row could consist of an "old" address which will not give us the first partition. We might want to return the slice from the listObjects method

Reply:
I've just wrapped the code to prevent exception during extract from empty data-frame. No change to the logic inside.